rskj-core/src/test/java/co/rsk/peg/federation/VoteFederationChangeTest.java

```diff
@@ -0,0 +1,672 @@
+
+        // Assert
+        // First call is successful
+        assertEquals(FederationChangeResponseCode.SUCCESSFUL.getCode(), firstVoteAddFederationResult);
+        // Second call fails
+        assertEquals(FederationChangeResponseCode.GENERIC_ERROR.getCode(), secondVoteAddFederationResult);
+
+        assertEquals(0, federationSupport.getPendingFederationSize());
+    }
+
+    @ParameterizedTest
+    @MethodSource("keysWithOneDifferentKeyArgProvider")
+    void voteAddFederatorPublicKeys_whenVotingAnotherFederatorThatSharesOneKeyWithExistingFederator_returnsFederatorAlreadyPresentResponseCode(
+        BtcECKey federator2BtcPublicKey, ECKey federator2RskPublicKey, ECKey federator2MstPublicKey
+    ) {
+        voteAndAssertCreateEmptyPendingFederation();
+
+        // Act
+        // Add first federator to pending federation
+        voteAndAssertAddFederatorPublicKeysToPendingFederation(federatorBtcKey, federatorRskKey, federatorMstKey);
+        // Voting new federator public keys,
+        // that will be considered the same as the previous one
+        // because they share at least one key
+        int firstVoteAddMultiFederator2KeysResult = voteToAddFederatorPublicKeysToPendingFederation(firstAuthorizedTx, federator2BtcPublicKey, federator2RskPublicKey, federator2MstPublicKey);
+
+        // Assert
+        assertEquals(FederationChangeResponseCode.FEDERATOR_ALREADY_PRESENT.getCode(), firstVoteAddMultiFederator2KeysResult);
+
+        // Pending federation size is 1, because authorizers first voted for the same federator and the second was ignored
+        assertEquals(1, federationSupport.getPendingFederationSize());
+    }
+
+    @ParameterizedTest
+    @MethodSource("keysWithOneDifferentKeyArgProvider")
+    void voteAddFederatorPublicKeys_whenVotingFederatorsWithOneDifferentKey_returnsSuccessfulButDoesNotAddFederator(
+        BtcECKey federatorBtcPublicKey, ECKey federatorRskPublicKey, ECKey federatorMstPublicKey
+    ) {
+        voteAndAssertCreateEmptyPendingFederation();
+
+        // Act
+        int firstVoteResult = voteToAddFederatorPublicKeysToPendingFederation(firstAuthorizedTx, federatorBtcKey, federatorRskKey, federatorMstKey);
+        // Voting a federator with just one different key from previous one.
+        // It will be considered a different federator,
+        // so second vote won't add it to the pending federation
+        int secondVoteResult = voteToAddFederatorPublicKeysToPendingFederation(secondAuthorizedTx, federatorBtcPublicKey, federatorRskPublicKey, federatorMstPublicKey);
+
+        // Assert
+        assertEquals(FederationChangeResponseCode.SUCCESSFUL.getCode(), firstVoteResult);
+        assertEquals(FederationChangeResponseCode.SUCCESSFUL.getCode(), secondVoteResult);
+
+        // Pending federation size is 0, because we voted for different federators
+        assertEquals(0, federationSupport.getPendingFederationSize());
+    }
+
```

Comment:
these two tests show an interesting inconsistent behavior:
`federator1` has keys btcKey1, rskKey1, mstKey1
`federator2` has keys btcKey1, rskKey1, mstKey2 (or btcKey1, rskKey2, mstKey1, or btcKey2, rskKey1, mstKey1)

if you vote first for `federator1` and second for `federator2`, it does not consider it as voting for the same federator.
But if `federator1` is already part of pending federation, voting for `federator2` returns `FEDERATOR_ALREADY_PRESENT` response code.